### PR TITLE
The animation control related changes

### PR DESCRIPTION
**Pull request by thesiv**

### CODE REVIEW

File: src/xrc/xh_animatctrl.cpp [hunk 1]
```diff
@@ -31,16 +34,67 @@ wxAnimationCtrlXmlHandler::wxAnimationCtrlXmlHandler() : wxXmlResourceHandler()
...
+        }
+    }
+    if ( GetBool("hidden", 0) == 1 )
+        ctrl->Hide();
 
-    wxScopedPtr<wxAnimation> animation(GetAnimation(wxT("animation")));
+    const wxString aniParam = "animation";
+    wxString path = GetParamValue(GetParamNode(aniParam));
+    if ( m_resource->GetFlags() & wxXRC_USE_ENVVARS )
+        path = wxExpandEnvVars(path);
```
Comment (vadz) [inline src/xrc/xh_animatctrl.cpp:68]: What's wrong with using `GetFilePath()` here?

File: samples/xrc/myframe.cpp [hunk 2]
```diff
@@ -206,14 +207,17 @@ void MyFrame::OnAnimationCtrlPlay(wxCommandEvent& event)
 
     wxWindow *win = btn->GetParent();
     wxAnimationCtrl *ctrl = XRCCTRL(*win, "controls_animation_ctrl", wxAnimationCtrl);
+    wxGenericAnimationCtrl *generic =
+        XRCCTRL(*win, "controls_generic_animation_ctrl", wxGenericAnimationCtrl);
```
Comment (vadz) [inline samples/xrc/myframe.cpp:211]: I'd probably take this inside `#ifdef wxHAS_NATIVE_ANIMATIONCTRL` as otherwise it's one and the same thing anyhow...

**Review by vadz — COMMENTED** [review]
Thanks!

Just one small problem: I'd like to understand why do we duplicate `wxXmlResourceHandlerImpl::GetFilePath()` code instead of reusing it?

**Comment by thesiv** [comment]
Fixed.

**Pull request merged**